DOC.md

```diff
@@ -1095,6 +1095,31 @@ hitting `<Tab>` while in Visualmode will populate the `*SELECT*`-vars for the ne
 snippet and then clear them.
 
 
+You can also add your own variables by using the `ls.extend_env(namespace, vars)` where:
+
+* `namespace` is string that does not contains the character "_".
+* `vars` 
+  * is a function that receives a string and returns a value for the var with that name
+  * or a table from var name to a value 
+    (in this case, if the value is a function it will be executed  lazily once per snippet expansion)
+
+A simple example to make it more clear:
+
+```lua
+local random_lang()
+    return return ({"LUA", "VIML"})[math.floor(math.random()+0.5)]
```

Comment:
btw, is it intentional that "VIML" will never be returned here? xD
Honestly, leave it like this, that's amazing :smile:

Reply:
crap, just noticed the double return :smile:

Reply:
I'm 100% for {"C", "LUA", "VIML9"} with the same `+1.5` :P
Putting some oil on the fire xD

Reply:
Rust is missing as well